azext_iot/tests/iothub/configurations/test_iot_config_unit.py

```diff
@@ -1023,3 +1030,64 @@ def test_config_apply_edge_error(
                 hub_name=mock_target["entity"],
                 content=sample_config_edge_malformed,
             )
+
+
+class TestConfigExport:
+    @pytest.fixture(params=[200])
+    def serviceclient(self, mocker, fixture_ghcs, fixture_sas, request, sample_config_read):
+        service_client = mocker.patch(path_service_client)
+        service_client.side_effect = [
+            build_mock_response(mocker, payload=[{"moduleId" : "$edgeAgent"}, {"moduleId" : "AzureBlobStorageonIoTEdge"}]),
```

Comment:
Can we please use the `mocked_response` fixture over `build_mock_response()`? We are trying to minimize usage of build_mock_response(), its on the deprecation to removal path.

Reply:
I see, does that mean we should minimize the usage of side_effect as well?😀

Reply:
In general usage of side_effect is OK, but path_service_client and build_mock_response are not guaranteed, an underlying change to azure client infra could break this pattern, usage of responses / mocked_response gives us more insulation since there is significant reliance on the `requests` library.